packages/lib/utils/url.ts

```diff
@@ -22,3 +22,19 @@ export const testURLmatch = (
       throw new Error("Invalid match type");
   }
 };
+
+// Helper function to validate callback URLs
+export const isValidCallbackUrl = (url: string, WEBAPP_URL: string): boolean => {
+  try {
+    const parsedUrl = new URL(url);
+    const allowedSchemes = ["https:", "http:"];
+
+    // Extract the domain from WEBAPP_URL
+    const parsedWebAppUrl = new URL(WEBAPP_URL);
+    const allowedDomains = [parsedWebAppUrl.hostname];
+
```

Comment:
Can you please also add one suspicious pattern check here?
``` typescript
// Check for suspicious patterns in URL
      const suspiciousPatterns = [
        /<script/i,
        /javascript:/i,
        /data:/i,
        /vbscript:/i,
        /onload=/i,
        /onerror=/i
      ];

      for (const pattern of suspiciousPatterns) {
        if (pattern.test(url)) {
          console.log(`Suspicious pattern found: ${pattern}`);
          return false;
        }
      }
```
Got the above code snippet from claude, seems helpful for enhanced security checks.

Reply:
I think we do not need it, because The `new URL(url)` constructor will throw an error if the input is not a valid URL. For example, strings containing HTML tags, JavaScript code, or malformed URLs will cause an exception, leading that function to return false 🤔

Reply:
Ohh okay, makes sense 👍